config/crd/bases/config.nri_balloonspolicies.yaml

```diff
@@ -147,6 +159,8 @@ spec:
                       type: string
                   required:
                   - name
+                  - preferCloseToDevices
```

Comment:
`preferCloseToDevices` is fully optional preference in balloon types. Does listing it in `required` make it mandatory in some validation?

I would hide `preferFarFromDevices` from CRDs.

Reply:
> `preferCloseToDevices` is fully optional preference in balloon types. Does listing it in `required` make it mandatory in some validation?

Required means, when user creates CR, it will be mandatory to set `preferCloseToDevices `to some value. If there is default value, then it the default value will be used if the user doesn't set it. But you don't have default here, so user must set it.

Reply:
Since [preferCloseToDevices ](https://github.com/containers/nri-plugins/blob/2308d1f5a296283e4b0ec0ad0fbdf841311dcebb/pkg/apis/config/v1alpha1/resmgr/policy/balloons/config.go#L191 ) is omitempty, I think it should be under optional in the CRD YAML. I wonder if our CRD generation has a bug or did you add it manually @klihub ?

Reply:
Yes, I think so, too. Thanks for spotting that.

Reply:
Actually, no. `omitempty` would imply `// +optional` had it been correctly added to the tags like it was for the rest of the struct members.

Reply:
Yes, correct. We need `// +optional` marker if we want the field to be optional.

Reply:
Okay, so I chose to tag `PreferFarFromDevices` with `"json:-"` to prevent it from being marshalled or unmarshalled from a CRD to the actual configuration. That's the most proper way I could come up with. The alternative maybe would have been to add a validation to force it to be `false`, but I think this is both simpler and better.